ui/design-system-components/src/components/Entry/EntryCard/index.tsx

```diff
@@ -1,130 +1,216 @@
+              }
+              label={
+                authorProfile.status === 'error' ? entryData.author.id : authorProfile.data.name
+              }
+              avatarImage={authorProfile.status === 'error' ? null : authorProfile.data?.avatar}
+              href={`${profileAnchorLink}/${entryData.author.id}`}
+              onClick={event => {
+                event.preventDefault();
+                if (onAvatarClick) onAvatarClick(entryData.author.id);
+              }}
+              ref={profileRef}
+            />
+          )}
+        </>
+        <Stack direction="row" spacing="gap-2" align="center" customStyle="shrink-0">
+          {entryData?.createdAt && !hidePublishTime && (
+            <Tooltip placement={'top'} content={formatDate(entryData?.createdAt, locale)}>
+              <Text customStyle="flex shrink-0 text(grey4 dark:grey7)">
+                {formatRelativeTime(entryData?.createdAt, locale)}
+              </Text>
+            </Tooltip>
+          )}
+          {entryData?.createdAt && (
+            <Tooltip
+              placement={'top'}
+              content={`${editedLabel} ${formatRelativeTime(entryData?.createdAt, locale)}`}
+            >
+              <Icon size="sm" type="PencilIcon" />
+            </Tooltip>
+          )}
+          {!entryData.active && (
+            <CardHeaderMenuDropdown
+              disabled={disableActions}
+              menuItems={
+                !entryData.author.isViewer
+                  ? [
+                      {
+                        icon: 'FlagIcon',
+                        handler: onEntryFlag,
+                        label: flagAsLabel,
+                        disabled: disableReporting,
+                      },
+                    ]
+                  : []
+              }
+              headerMenuExt={headerMenuExt}
+            />
+          )}
+        </Stack>
+      </Stack>
+      {isRemoved && (
+        <EntryCardRemoved
+          isAuthor={entryData.author.isViewer}
+          removedByAuthorLabel={removedByAuthorLabel}
+          removedByMeLabel={removedByMeLabel}
+        />
+      )}
+      {!isRemoved && (
+        <Button onClick={onContentClick} plain>
+          <Stack
+            customStyle={`px-4 max-h-[50rem] ${
+              scrollHiddenContent ? 'overflow-auto' : 'overflow-hidden'
+            } ${contentClickable ? 'cursor-pointer' : 'cursor-default'}`}
+            data-testid="entry-content"
+          >
+            {rest.itemType === EntityTypes.REFLECT ? (
+              <ReadOnlyEditor
+                content={rest.slateContent}
+                handleMentionClick={rest.onMentionClick}
+                handleTagClick={rest.onTagClick}
+                handleLinkClick={url => {
+                  rest.navigateTo?.({ getNavigationUrl: () => url });
+                }}
+              />
+            ) : (
+              rest.sortedContents.map(item => (
+                <Extension
```

Comment:
We should not have this <Extension /> here, we should have {props.children} instead and control the extension from outside. I'm working to replace this extension component with another one which will be in it's own library (like the feed lib)